x-pack/plugins/security_solution/public/timelines/components/edit_data_provider/components/controlled_combobox_input.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useMemo, useEffect, useCallback } from 'react';
+
+import type { EuiComboBoxOptionOption } from '@elastic/eui';
+import { EuiComboBox } from '@elastic/eui';
+
+import type { DataProviderType } from '../../timeline/data_providers/data_provider';
+import { isStringOrNumberArray } from '../../timeline/helpers';
+import { isValueFieldInvalid } from '../helpers';
+import * as i18n from '../translations';
+
+interface ControlledDataProviderInput {
+  disableButtonCallback: (disableButton: boolean) => void;
+  onChangeCallback: (value: string | number | string[]) => void;
+  type: DataProviderType;
+  value: string | number | Array<string | number>;
+}
+
+export const ControlledComboboxInput = ({
+  value,
+  onChangeCallback,
+  type,
+  disableButtonCallback,
+}: ControlledDataProviderInput) => {
+  const [includeValues, setIncludeValues] = useState(convertValuesToComboboxValueArray(value));
+
+  const isInvalid = useMemo(
+    () => includeValues.every((item) => isValueFieldInvalid(type, item.label)),
```

Comment:
consider replacing the use of `every` here if it's sufficient for just one field to be invalid

Reply:
changed!